### ISSUE TITLE
Cairo installation issue

### ISSUE THREAD

**Issue by MontrealMike** [JuliaGraphics/Cairo.jl#302]
I am trying to get the _ImageView_ package to work.  The  Pkg.add works OK but I get a failure at the using step.  I believe I have traced the issue to the Cairo package build and a dependency on _gettext_. Any help is appreciated.

__________________ System Info __________________
OS: Linux Ubuntu 19.04 (64 bit)

julia> versioninfo()
Julia Version 1.0.4
Commit 38e9fb7f80 (2019-05-16 03:38 UTC)
Platform Info:
  OS: Linux (i686-pc-linux-gnu)
  CPU: Intel(R) Core(TM) i7-7567U CPU @ 3.50GHz
  WORD_SIZE: 32
  LIBM: libopenlibm
  LLVM: libLLVM-6.0.0 (ORCJIT, skylake)

_______________ Cairo Build Output _________________
(v1.0) pkg> build Cairo
  Building Homebrew → `~/.julia/packages/Homebrew/s09IX/deps/build.log`
  Building LibCURL ─→ `~/.julia/packages/LibCURL/lWJxD/deps/build.log`
  Building WinRPM ──→ `~/.julia/packages/WinRPM/Y9QdZ/deps/build.log`
  Building Cairo ───→ `~/.julia/packages/Cairo/p68X8/deps/build.log`
[sudo] password for michael: 
┌ Error: Error building `Cairo`: 
│ Reading package lists...
│ Building dependency tree...
│ Reading state information...
│ gettext is already the newest version (0.19.8.1-9).
│ 0 upgraded, 0 newly installed, 0 to remove and 11 not upgraded.
│ ERROR: LoadError: Provider AptGet failed to satisfy dependency gettext
│ Stacktrace:
│  [1] error(::String) at ./error.jl:33
│  [2] satisfy!(::BinDeps.LibraryDependency, ::Array{DataType,1}) at /home/michael/.julia/packages/BinDeps/ZEval/src/dependencies.jl:945
│  [3] satisfy!(::BinDeps.LibraryGroup, ::Array{DataType,1}) at /home/michael/.julia/packages/BinDeps/ZEval/src/dependencies.jl:910
│  [4] satisfy!(::BinDeps.LibraryGroup) at /home/michael/.julia/packages/BinDeps/ZEval/src/dependencies.jl:874
│  [5] top-level scope at /home/michael/.julia/packages/BinDeps/ZEval/src/dependencies.jl:977
│  [6] include at ./boot.jl:317 [inlined]
│  [7] include_relative(::Module, ::String) at ./loading.jl:1044
│  [8] include(::Module, ::String) at ./sysimg.jl:29
│  [9] include(::String) at ./client.jl:392
│  [10] top-level scope at none:0
│ in expression starting at /home/michael/.julia/packages/Cairo/p68X8/deps/build.jl:176
│ Installing dependency gettext via `sudo apt-get install gettext`:
└ @ Pkg.Operations /buildworker/worker/package_linux32/build/usr/share/julia/stdlib/v1.0/Pkg/src/Operations.jl:1096

**Comment by giordano** [comment]
Please, update to the latest version of Cairo, 0.7, that should solve the issue.

**Comment by MontrealMike** [comment]
Thanks Giordano.  I have updated Cairo to 0.7 but am getting the same result.  I am not sure what to try next. 

_______________ Cairo.jl code _________________
~/.julia/packages/Cairo/p68X8$ head src/Cairo.jl 
```
VERSION < v"0.7.0-beta2.199" && __precompile__()

module Cairo

using Compat
import Compat.Sys

depsjl = joinpath(dirname(@__FILE__), "..", "deps", "deps.jl")
isfile(depsjl) ? include(depsjl) : error("Cairo not properly ",
    "installed. Please run\nPkg.build(\"Cairo\")")
```


_______________ Cairo Build Output _________________
(v1.0) pkg> build Cairo
  Building Homebrew → `~/.julia/packages/Homebrew/s09IX/deps/build.log`
  Building LibCURL ─→ `~/.julia/packages/LibCURL/lWJxD/deps/build.log`
  Building WinRPM ──→ `~/.julia/packages/WinRPM/Y9QdZ/deps/build.log`
  Building Cairo ───→ `~/.julia/packages/Cairo/p68X8/deps/build.log`
┌ Error: Error building `Cairo`: 
│ Reading package lists...
│ Building dependency tree...
│ Reading state information...
│ gettext is already the newest version (0.19.8.1-9).
│ 0 upgraded, 0 newly installed, 0 to remove and 11 not upgraded.
│ ERROR: LoadError: Provider AptGet failed to satisfy dependency gettext
│ Stacktrace:
│  [1] error(::String) at ./error.jl:33
│  [2] satisfy!(::BinDeps.LibraryDependency, ::Array{DataType,1}) at /home/michael/.julia/packages/BinDeps/ZEval/src/dependencies.jl:945
│  [3] satisfy!(::BinDeps.LibraryGroup, ::Array{DataType,1}) at /home/michael/.julia/packages/BinDeps/ZEval/src/dependencies.jl:910
│  [4] satisfy!(::BinDeps.LibraryGroup) at /home/michael/.julia/packages/BinDeps/ZEval/src/dependencies.jl:874
│  [5] top-level scope at /home/michael/.julia/packages/BinDeps/ZEval/src/dependencies.jl:977
│  [6] include at ./boot.jl:317 [inlined]
│  [7] include_relative(::Module, ::String) at ./loading.jl:1044
│  [8] include(::Module, ::String) at ./sysimg.jl:29
│  [9] include(::String) at ./client.jl:392
│  [10] top-level scope at none:0
│ in expression starting at /home/michael/.julia/packages/Cairo/p68X8/deps/build.jl:176
│ Installing dependency gettext via `sudo apt-get install gettext`:
└ @ Pkg.Operations /buildworker/worker/package_linux32/build/usr/share/julia/stdlib/v1.0/Pkg/src/Operations.jl:1096

**Comment by giordano** [comment]
You're not using Cairo 0.7 if it's installing `Homebrew` and `WinRPM`.  Do
```
pkg> status Cairo
```
to see what version of Cairo you have.  Probably there are other packages holding it back because of incompatibilities?  Try running
```
pkg> add Cairo@0.7
```
and report the error message.

Please, use code formatting when you want to show code here, just select the code and click on the `<>` button

**Comment by MontrealMike** [comment]
Interesting ... obviously I am very new to this and I appreciate your patience.  It looks like my issue may now be with Gtk if I interpret the output below correctly.  

```
(v1.0) pkg> status Cairo
ERROR: Wrong number of arguments

(v1.0) pkg> add Cairo@0.7
 Resolving package versions...
ERROR: Unsatisfiable requirements detected for package Cairo [159f3aea]:
 Cairo [159f3aea] log:
 ├─possible versions are: [0.0.1, 0.2.0-0.2.35, 0.3.0-0.3.1, 0.4.0, 0.5.0-0.5.6, 0.6.0, 0.7.0] or uninstalled
 ├─restricted to versions 0.7 by an explicit requirement, leaving only versions 0.7.0
 └─restricted by compatibility requirements with Gtk [4c0ca9eb] to versions: [0.2.2-0.2.35, 0.3.0-0.3.1, 0.4.0, 0.5.0-0.5.6, 0.6.0] — no versions left
   └─Gtk [4c0ca9eb] log:
     ├─possible versions are: [0.1.0, 0.2.0, 0.3.0-0.3.2, 0.4.0, 0.5.0, 0.6.0-0.6.2, 0.7.0-0.7.10, 0.8.0-0.8.6, 0.9.0-0.9.3, 0.10.0-0.10.4, 0.11.0, 0.12.0, 0.13.0-0.13.1, 0.14.0, 0.16.0-0.16.5, 0.17.0] or uninstalled
     └─restricted to versions 0.17.0 by an explicit requirement, leaving only versions 0.17.0


```

**Comment by giordano** [comment]
Right, you need gtk for imageview. Ok, version 0.7 of Cairo is indeed incompatible with any released version of GTK, we're in the process of improving the installation experience for the users.

What was exactly the problem at the beginning? Were you prompted for the password to install the library? Didn't it work?

**Comment by MontrealMike** [comment]
I have deleted ImageView and dependencied using:
```
(v1.0) pkg> rm --manifest ImageView
  Updating `~/.julia/environments/v1.0/Project.toml`
  [86fae568] - ImageView v0.9.0
  Updating `~/.julia/environments/v1.0/Manifest.toml`
  [e1450e63] - BufferedStreams v1.0.0
  [159f3aea] - Cairo v0.6.0
  [4c0ca9eb] - Gtk v0.17.0
  [27996c0f] - GtkReactive v0.6.0
  [0862f596] - HTTPClient v0.2.1
  [d9be37ee] - Homebrew v0.7.1
  [86fae568] - ImageView v0.9.0
  [b27032c2] - LibCURL v0.5.2
  [522f3ed2] - LibExpat v0.5.0
  [2ec943e9] - Libz v1.0.0
  [a223df75] - Reactive v0.8.3
  [d5f540fe] - RoundingIntegers v0.2.0
  [c17dfb99] - WinRPM v0.4.2

(v1.0) pkg> gc
    Active manifests:
        `~/.julia/environments/v1.0/Manifest.toml`
   Deleted `~/.julia/packages/BufferedStreams/hCA7W` (102.235 KiB)
   Deleted `~/.julia/packages/Cairo/p68X8` (1.220 MiB)
   Deleted `~/.julia/packages/Gtk/aP55V` (3.977 MiB)
   Deleted `~/.julia/packages/GtkReactive/2h7NX` (327.676 KiB)
   Deleted `~/.julia/packages/HTTPClient/ZQR55` (37.669 KiB)
   Deleted `~/.julia/packages/Homebrew/s09IX` (62.268 KiB)
   Deleted `~/.julia/packages/ImageView/1uiRS` (500.141 KiB)
   Deleted `~/.julia/packages/LibCURL/lWJxD` (6.860 MiB)
   Deleted `~/.julia/packages/LibExpat/6jLDP` (127.247 KiB)
   Deleted `~/.julia/packages/Libz/zMAun` (80.744 KiB)
   Deleted `~/.julia/packages/Reactive/g1LZ6` (91.988 KiB)
   Deleted `~/.julia/packages/RoundingIntegers/5Xy7e` (11.827 KiB)
   Deleted `~/.julia/packages/WinRPM/Y9QdZ` (25.091 KiB)
   Deleted 13 package installations (13.392 MiB)

```
Now I add it in again. Note password entered after **Building Cairo**.
Just after this I get the error related to **gettext** even though gettext is installed.
Later after **Building Gtk** I get another error 
_ERROR: LoadError: Provider AptGet failed to satisfy dependency glib_

```
(v1.0) pkg> add ImageView
  Updating registry at `~/.julia/registries/General`
  Updating git-repo `https://github.com/JuliaRegistries/General.git`
 Resolving package versions...
 Installed RoundingIntegers ─ v0.2.0
 Installed LibExpat ───────── v0.5.0
 Installed LibCURL ────────── v0.5.2
 Installed BufferedStreams ── v1.0.0
 Installed Reactive ───────── v0.8.3
 Installed Gtk ────────────── v0.17.0
 Installed ImageView ──────── v0.9.0
 Installed WinRPM ─────────── v0.4.2
 Installed HTTPClient ─────── v0.2.1
 Installed Homebrew ───────── v0.7.1
 Installed Libz ───────────── v1.0.0
 Installed Cairo ──────────── v0.6.0
 Installed GtkReactive ────── v0.6.0
  Updating `~/.julia/environments/v1.0/Project.toml`
  [86fae568] + ImageView v0.9.0
  Updating `~/.julia/environments/v1.0/Manifest.toml`
  [e1450e63] + BufferedStreams v1.0.0
  [159f3aea] + Cairo v0.6.0
  [4c0ca9eb] + Gtk v0.17.0
  [27996c0f] + GtkReactive v0.6.0
  [0862f596] + HTTPClient v0.2.1
  [d9be37ee] + Homebrew v0.7.1
  [86fae568] + ImageView v0.9.0
  [b27032c2] + LibCURL v0.5.2
  [522f3ed2] + LibExpat v0.5.0
  [2ec943e9] + Libz v1.0.0
  [a223df75] + Reactive v0.8.3
  [d5f540fe] + RoundingIntegers v0.2.0
  [c17dfb99] + WinRPM v0.4.2
  Building Homebrew → `~/.julia/packages/Homebrew/s09IX/deps/build.log`
  Building LibCURL ─→ `~/.julia/packages/LibCURL/lWJxD/deps/build.log`
  Building WinRPM ──→ `~/.julia/packages/WinRPM/Y9QdZ/deps/build.log`
  Building Cairo ───→ `~/.julia/packages/Cairo/p68X8/deps/build.log`
[sudo] password for michael: 
┌ Error: Error building `Cairo`: 
│ Reading package lists...
│ Building dependency tree...
│ Reading state information...
│ gettext is already the newest version (0.19.8.1-9).
│ 0 upgraded, 0 newly installed, 0 to remove and 11 not upgraded.
│ ERROR: LoadError: Provider AptGet failed to satisfy dependency gettext
│ Stacktrace:
│  [1] error(::String) at ./error.jl:33
│  [2] satisfy!(::BinDeps.LibraryDependency, ::Array{DataType,1}) at /home/michael/.julia/packages/BinDeps/ZEval/src/dependencies.jl:945
│  [3] satisfy!(::BinDeps.LibraryGroup, ::Array{DataType,1}) at /home/michael/.julia/packages/BinDeps/ZEval/src/dependencies.jl:910
│  [4] satisfy!(::BinDeps.LibraryGroup) at /home/michael/.julia/packages/BinDeps/ZEval/src/dependencies.jl:874
│  [5] top-level scope at /home/michael/.julia/packages/BinDeps/ZEval/src/dependencies.jl:977
│  [6] include at ./boot.jl:317 [inlined]
│  [7] include_relative(::Module, ::String) at ./loading.jl:1044
│  [8] include(::Module, ::String) at ./sysimg.jl:29
│  [9] include(::String) at ./client.jl:392
│  [10] top-level scope at none:0
│ in expression starting at /home/michael/.julia/packages/Cairo/p68X8/deps/build.jl:176
│ Installing dependency gettext via `sudo apt-get install gettext`:
└ @ Pkg.Operations /buildworker/worker/package_linux32/build/usr/share/julia/stdlib/v1.0/Pkg/src/Operations.jl:1096
  Building Gtk ─────→ `~/.julia/packages/Gtk/aP55V/deps/build.log`
┌ Error: Error building `Gtk`: 
│ Reading package lists...
│ Building dependency tree...
│ Reading state information...
│ libgtk-3-dev is already the newest version (3.24.8-1ubuntu1).
│ 0 upgraded, 0 newly installed, 0 to remove and 11 not upgraded.
│ ERROR: LoadError: Provider AptGet failed to satisfy dependency glib
│ Stacktrace:
│  [1] error(::String) at ./error.jl:33
│  [2] satisfy!(::BinDeps.LibraryDependency, ::Array{DataType,1}) at /home/michael/.julia/packages/BinDeps/ZEval/src/dependencies.jl:945
│  [3] satisfy!(::BinDeps.LibraryGroup, ::Array{DataType,1}) at /home/michael/.julia/packages/BinDeps/ZEval/src/dependencies.jl:910
│  [4] satisfy!(::BinDeps.LibraryGroup) at /home/michael/.julia/packages/BinDeps/ZEval/src/dependencies.jl:874
│  [5] top-level scope at /home/michael/.julia/packages/BinDeps/ZEval/src/dependencies.jl:977
│  [6] include at ./boot.jl:317 [inlined]
│  [7] include_relative(::Module, ::String) at ./loading.jl:1044
│  [8] include(::Module, ::String) at ./sysimg.jl:29
│  [9] include(::String) at ./client.jl:392
│  [10] top-level scope at none:0
│ in expression starting at /home/michael/.julia/packages/Gtk/aP55V/deps/build.jl:63
│ Installing dependency libgtk-3-dev via `sudo apt-get install libgtk-3-dev`:
└ @ Pkg.Operations /buildworker/worker/package_linux32/build/usr/share/julia/stdlib/v1.0/Pkg/src/Operations.jl:1096


```

**Comment by tknopp** [comment]
Is this ubuntu?

**Comment by MontrealMike** [comment]
```
Processor		: Intel(R) Core(TM) i7-7567U CPU @ 3.50GHz
Operating System   : Ubuntu 19.04
```

**Comment by tknopp** [comment]
I think that this might be an issue with the latest ubuntu. One idea is to edit
https://github.com/JuliaGraphics/Cairo.jl/blob/de25db69a488ef868124085405a54490200fe8e2/deps/build.jl#L60
and comment out the "gettext" entry.

**Comment by tknopp** [comment]
That file is in ~/julia/packages/Cairo/???/deps

**Comment by MontrealMike** [comment]
Sorry.  That failed also.  Thanks for all your help so far.  I must sign off for today, but will be at it again tomorrow AM.

I attach the edited _build.jl_ as well as the _build.log_.

--------------------------- _build.jl_ ------------------------------------------------------------------------------
```
using BinDeps

using Compat
import Compat.Libdl
import Compat.Sys

if Sys.isapple()
    deps_file_str = open(joinpath(dirname(pathof(BinDeps)), "dependencies.jl")) do file
        read(file, String)
    end
    if occursin("ZEval",pathof(BinDeps))
        patched_dlclose = replace(deps_file_str, "Libdl.dlclose(h)" => "println(\"ignored: dlclose()\")")
        include_string(BinDeps,patched_dlclose)
    end
end


@BinDeps.setup

# check for cairo version
function validate_cairo_version(name,handle)
    f = Libdl.dlsym_e(handle, "cairo_version")
    f == C_NULL && return false
    v = ccall(f, Int32,())
    return v > 10800
end

group = library_group("cairo")


libpng = library_dependency("png", aliases = ["libpng","libpng-1.5.14","libpng15","libpng12.so.0","libpng12"], runtime = false, group = group)
pixman = library_dependency("pixman", aliases = ["libpixman","libpixman-1","libpixman-1-0","libpixman-1.0"], depends = [libpng], runtime = false, group = group)
libffi = library_dependency("ffi", aliases = ["libffi"], runtime = false, group = group)
gettext = library_dependency("gettext", aliases = ["libintl", "preloadable_libintl", "libgettextpo", "intltool"], os = :Unix, group = group)
gobject = library_dependency("gobject", aliases = ["libgobject-2.0-0", "libgobject-2.0", "libgobject-2_0-0", "libgobject-2.0.so.0"], depends=[libffi, gettext], group = group)
freetype = library_dependency("freetype", aliases = ["libfreetype"], runtime = false, group = group)
fontconfig = library_dependency("fontconfig", aliases = ["libfontconfig-1", "libfontconfig", "libfontconfig.so.1"], depends = [freetype], runtime = false, group = group)
cairo = library_dependency("cairo", aliases = ["libcairo-2", "libcairo","libcairo.so.2", "libcairo2"], depends = [gobject,fontconfig,libpng], group = group, validate = validate_cairo_version)
pango = library_dependency("pango", aliases = ["libpango-1.0-0", "libpango-1.0","libpango-1.0.so.0", "libpango-1_0-0"], group = group)
pangocairo = library_dependency("pangocairo", aliases = ["libpangocairo-1.0-0", "libpangocairo-1.0", "libpangocairo-1.0.so.0"], depends = [cairo], group = group)
zlib = library_dependency("zlib", aliases = ["libzlib","zlib1"], os = :Windows, group = group)

if Sys.iswindows()
    using WinRPM
    provides(WinRPM.RPM,"libpango-1_0-0",[pango,pangocairo],os = :Windows)
    provides(WinRPM.RPM,["glib2", "libgobject-2_0-0"],gobject,os = :Windows)
    provides(WinRPM.RPM,"zlib-devel",zlib,os = :Windows)
    provides(WinRPM.RPM,["libcairo2","libharfbuzz0"],cairo,os = :Windows)
end

if Sys.isapple()
    using Homebrew
    Homebrew.add("graphite2")
    provides( Homebrew.HB, "cairo", cairo, os = :Darwin )
    provides( Homebrew.HB, "pango", [pango, pangocairo], os = :Darwin, onload =
    """
    function __init__()
        ENV["PANGO_SYSCONFDIR"] = joinpath("$(Homebrew.prefix())", "etc")
    end
    """ )
    provides( Homebrew.HB, "fontconfig", fontconfig, os = :Darwin )
    provides( Homebrew.HB, "glib", gobject, os = :Darwin )
    provides( Homebrew.HB, "libpng", libpng, os = :Darwin )
    provides( Homebrew.HB, "gettext", gettext, os = :Darwin )
    provides( Homebrew.HB, "freetype", freetype, os = :Darwin )
    provides( Homebrew.HB, "libffi", libffi, os = :Darwin )
    provides( Homebrew.HB, "pixman", pixman, os = :Darwin )
end

# System Package Managers
provides(AptGet,
    Dict(
        "libcairo2" => cairo,
        "libfontconfig1" => fontconfig,
        "libpango1.0-0" => [pango,pangocairo],
        "libglib2.0-0" => gobject,
        "libpng12-0" => libpng,
        "libpixman-1-0" => pixman
#        "gettext" => gettext
    ))

# TODO: check whether these are accurate
provides(Yum,
    Dict(
        "cairo" => cairo,
        "fontconfig" => fontconfig,
        "pango" => [pango,pangocairo],
        "glib2" => gobject,
        "libpng" => libpng,
        "gettext-libs" => gettext
    ))

provides(Zypper,
    Dict(
        "libcairo2" => cairo,
        "libfontconfig" => fontconfig,
        "libpango-1_0" => [pango,pangocairo],
        "libglib-2_0" => gobject,
        "libpng12" => libpng,
        "libpixman-1" => pixman,
        "gettext" => gettext
    ))

const png_version = "1.5.14"

provides(Sources,
    Dict(
        URI("http://www.cairographics.org/releases/pixman-0.28.2.tar.gz") => pixman,
        URI("http://www.cairographics.org/releases/cairo-1.12.16.tar.xz") => cairo,
        URI("http://download.savannah.gnu.org/releases/freetype/freetype-2.4.11.tar.gz") => freetype,
        URI("http://www.freedesktop.org/software/fontconfig/release/fontconfig-2.10.2.tar.gz") => fontconfig,
        URI("http://ftp.gnu.org/pub/gnu/gettext/gettext-0.18.2.tar.gz") => gettext,
        URI("ftp://ftp.simplesystems.org/pub/libpng/png/src/history/libpng15/libpng-$(png_version).tar.gz") => libpng,
        URI("ftp://sourceware.org/pub/libffi/libffi-3.0.11.tar.gz") => libffi,
        URI("http://ftp.gnome.org/pub/gnome/sources/glib/2.34/glib-2.34.3.tar.xz") => gobject,
        URI("http://ftp.gnome.org/pub/GNOME/sources/pango/1.32/pango-1.32.6.tar.xz") => [pango,pangocairo],
        URI("http://zlib.net/zlib-1.2.7.tar.gz") => zlib
    ))

xx(t...) = (Sys.iswindows() ? t[1] : (Sys.islinux() || length(t) == 2) ? t[2] : t[3])

provides(BuildProcess,
    Dict(
        Autotools(libtarget = "pixman/libpixman-1.la", installed_libname = xx("libpixman-1-0.","libpixman-1.","libpixman-1.0.")*Libdl.dlext) => pixman,
        Autotools(libtarget = xx("objs/.libs/libfreetype.la","libfreetype.la")) => freetype,
        Autotools(libtarget = "src/libfontconfig.la") => fontconfig,
        Autotools(libtarget = "src/libcairo.la", configure_options = append!(append!(
                AbstractString[],
                !Sys.islinux() ? AbstractString["--without-x","--disable-xlib","--disable-xcb"] : AbstractString[]),
                Sys.isapple() ? AbstractString["--enable-quartz","--enable-quartz-font","--enable-quartz-image","--disable-gl"] : AbstractString[])) => cairo,
        Autotools(libtarget = "gettext-tools/gnulib-lib/.libs/libgettextlib.la") => gettext,
        Autotools(libtarget = "libffi.la") => libffi,
        Autotools(libtarget = "gobject/libgobject-2.0.la") => gobject,
        Autotools(libtarget = "pango/libpango-1.0.la") => [pango,pangocairo]
    ))

provides(BuildProcess,Autotools(libtarget = "libpng15.la"),libpng,os = :Unix)


if VERSION < v"1.0.0"
provides(SimpleBuild,
    (@build_steps begin
        GetSources(zlib)
        @build_steps begin
            ChangeDirectory(joinpath(BinDeps.depsdir(zlib),"src","zlib-1.2.7"))
            MakeTargets(["-fwin32/Makefile.gcc"])
            #MakeTargets(["-fwin32/Makefile.gcc","DESTDIR=../../usr/","INCLUDE_PATH=include","LIBRARY_PATH=lib","SHARED_MODE=1","install"])
        end
    end),zlib, os = :Windows)
end

prefix=joinpath(BinDeps.depsdir(libpng),"usr")
uprefix = replace(replace(prefix,"\\" => "/"),"C:/" => "/c/")
pngsrcdir = joinpath(BinDeps.depsdir(libpng),"src","libpng-$png_version")
pngbuilddir = joinpath(BinDeps.depsdir(libpng),"builds","libpng-$png_version")
provides(BuildProcess,
    (@build_steps begin
        GetSources(libpng)
        CreateDirectory(pngbuilddir)
        @build_steps begin
            ChangeDirectory(pngbuilddir)
            FileRule(joinpath(prefix,"lib","libpng15.dll"),@build_steps begin
                `cmake -DCMAKE_INSTALL_PREFIX="$prefix" -G"MSYS Makefiles" $pngsrcdir`
                `make`
                `cp 'libpng*.dll' $prefix/lib`
                `cp 'libpng*.a' $prefix/lib`
                `cp 'libpng*.pc' $prefix/lib/pkgconfig`
                `cp pnglibconf.h $prefix/include`
                `cp $pngsrcdir/png.h $prefix/include`
                `cp $pngsrcdir/pngconf.h $prefix/include`
            end)
        end
    end),libpng, os = :Windows)


@BinDeps.install Dict([(:gobject, :_jl_libgobject),
                       (:cairo, :_jl_libcairo),
                       (:pango, :_jl_libpango),
                       (:pangocairo, :_jl_libpangocairo)])


if Sys.isapple()
    import Pkg.Types.printpkgstyle
    printpkgstyle(stdout, :Installed, "Patched install script, restarting Julia is recommended.")
end

```

**Comment by MontrealMike** [comment]
---------------------------- _build.log_ -----------------------------------------------------------
[build.log](https://github.com/JuliaGraphics/Cairo.jl/files/3688168/build.log)

**Comment by MontrealMike** [comment]
Just an update for anyone interested.  I have not resolved this problem.  Instead I am going back to my comfort zone which is R.
I plan to install the CallJulia package which allows R to call Julia functions.  This gives me the speed of Julia where needed and the R graphics which are great (and which work on my machine).  Anyway, that's the plan.
I will keep an eye out for any news on getting Cairo/Gtk working on recent Ubuntu versions.
Thanks to all for their help.

**Comment by MontrealMike** [comment]
Problem solved.

The issue was that I was using the 32 bit version of julia.  This is what gets installed from the Ubuntu repositories even though I am using 64 bit Ubuntu.  The only way I could find to to get the 64 bit version is to do it straight from the Julia web site.  Once I did this everything worked as advertised.

Thanks to @giordano and @tknopp for your help.

**Comment by tknopp** [comment]
Actually one should not use the Julia version that is bundled with the distros. So downloading from the Julia website is the correct thing to do.

**Comment by giordano** [comment]
@MontrealMike did you get Julia from snapstore?  Today another user got into troubles because installed the wrong version of Julia in Ubuntu: https://discourse.julialang.org/t/sys-word-size-gives-wrong-bit-size/29879

**Comment by MontrealMike** [comment]
Yes I did, The store offers three julia downloads:

1. julia
2. julia-stable
3. julia-mrcinv

There is no information on how each one differs from the other.  I tried the first 2 and got the same result.  Something like:

```
julia> versioninfo()
Julia Version 1.0.4
Commit 38e9fb7f80 (2019-05-16 03:38 UTC)
Platform Info:
OS: Linux (i686-pc-linux-gnu)
CPU: Intel(R) Core(TM) i7-7567U CPU @ 3.50GHz
WORD_SIZE: 32
LIBM: libopenlibm
LLVM: libLLVM-6.0.0 (ORCJIT, skylake)
```

Note the 32 bit WORD_SIZE.  As mentioned above, my particular issue was solved when I obtained julia straight from the Julia website.  This one had a WORD_SIZE of 64.
